client/sites-dashboard/hooks/use-sites-dashboard-import-site-url.ts

```diff
@@ -1,26 +1,18 @@
 import { Primitive } from 'utility-types';
+import { useIsCurrentlyHostingFlow } from 'calypso/landing/stepper/utils/hosting-flow';
 import { addQueryArgs } from 'calypso/lib/url';
-import { useSelector } from 'calypso/state';
-import getCurrentQueryArguments from 'calypso/state/selectors/get-current-query-arguments';
-import getUserSetting from 'calypso/state/selectors/get-user-setting';
-import { AppState } from 'calypso/types';
 import { TRACK_SOURCE_NAME } from '../utils';
 
 export const useSitesDashboardImportSiteUrl = (
 	additionalParameters: Record< string, Primitive >
 ) => {
-	const isDevAccount = useSelector( ( state ) => getUserSetting( state, 'is_dev_account' ) );
-
-	const isHostingFlow = useSelector(
-		( state: AppState ) => getCurrentQueryArguments( state )?.[ 'hosting-flow' ] === 'true'
-	);
+	const isHostingFlow = useIsCurrentlyHostingFlow();
 
 	return addQueryArgs(
 		{
 			source: TRACK_SOURCE_NAME,
 			...additionalParameters,
-			'hosting-flow': isHostingFlow ? true : null,
 		},
-		isDevAccount || isHostingFlow ? '/setup/import-hosted-site' : '/start/import'
+		isHostingFlow ? '/setup/import-hosted-site' : '/start/import'
```

Comment:
When coming from `/start/hosting`, which basically means you came from `/hosting` and clicked a CTA.

The user will see the "Let's create a site" page and they will be able to import the site without an already existing site there.

Reply:
That's because this hook is used in the `/sites` page, which gets an overlay in case the user comes from `/start/hosting`. If there is no overlay, then we render the `/start/import` URL under the multi-action top-right corner CTA.